File: 2121-intervals-between-identical-elements/2121-intervals-between-identical-elements.py
class Solution:
    def getDistances(self, arr: List[int]) -> List[int]:
        cntMap, sumMap = {}, {}
        n = len(arr)
        res = [0 for i in range(n)]
        for i in range(n):
            res[i] += cntMap.get(arr[i],0)*i - (sumMap.get(arr[i],0))
            cntMap[arr[i]] = cntMap.get(arr[i],0) + 1
            sumMap[arr[i]] = sumMap.get(arr[i],0) + i
        cntMap.clear()
        sumMap.clear()
        for i in range(n-1, -1, -1):
            res[i] += sumMap.get(arr[i],0) - (cntMap.get(arr[i],0)*i)
            cntMap[arr[i]] = cntMap.get(arr[i],0) + 1
            sumMap[arr[i]] = sumMap.get(arr[i],0) + i
        return res

File: 2121-intervals-between-identical-elements/2121-intervals-between-identical-elements.java
class Solution {
    public long[] getDistances(int[] arr) {
        long[] res = new long[arr.length];
        HashMap<Integer, Long> countMap = new HashMap<>(), sumMap = new HashMap<>();
        for(int i=0; i<arr.length; i++){
            res[i] += (long)((countMap.getOrDefault(arr[i], 0L)*i*1L - (sumMap.getOrDefault(arr[i], 0L))));
            countMap.put(arr[i], countMap.getOrDefault(arr[i], 0L)+1);
            sumMap.put(arr[i], sumMap.getOrDefault(arr[i],0L)+i);
        }
        sumMap.clear();
        countMap.clear();
        for(int i=arr.length-1; i>=0; i--){
            res[i] += (long)(sumMap.getOrDefault(arr[i], 0L) - countMap.getOrDefault(arr[i], 0L)*i*1L);
            countMap.put(arr[i], countMap.getOrDefault(arr[i], 0L)+1);
            sumMap.put(arr[i], sumMap.getOrDefault(arr[i],0L)+i);
        }
        return res;
    }
}